Topics/Tech_Stacks/Cypress.md

```diff
@@ -0,0 +1,65 @@
+## E2E Testing with Cypress
+
+# Cypress Introduction
+
+Cypress is mainly used for testing web applications, especially those built in javascript. It provides an interface to programatically test your application, and visually what went wrong (or right) in tests. 
+  
+# Why do end to end testing?  
+  
+[https://circleci.com/blog/what-is-end-to-end-testing/](https://circleci.com/blog/what-is-end-to-end-testing/)
+
+The above link has a good explanation on what end to end testing is and why it should be used. 
+
+Cypress very closely mimics a real user, think of it as a robot accessing your website from a browser like a human would, but you can program the robot to interact with your website however you like and programmatically check the output on the screen.
+
+# Installation and setup:  
+  
+Cypress can be automatically installed with npm: `npm install cypress`
```

Comment:
For readers who may be unfamiliar with npm, it could be helpful to quickly describe why it is used for installation.

Reply:
I don't exactly agree with this, because npm is just so important for javascript and node.js that its even bundled with node, but I added a hyperlink for npm just in case.

Reply:
True, a hyperlink for npm is probably good enough.